### DISCUSSION TITLE
Where is the function to update items in the player's vault?

### DISCUSSION THREAD

**Discussion by Fanya-dev** [SmartlyDressedGames/Unturned-3.x-Community#3875] [General]
The storage has a clear() function that simply calls Clear on the array, items from the storage are gone but the space is taken up on the client side, how do I update the data on the storage?

**Comment by sunnamed434** [comment]
I'm doing smth like this, and before calling the `CloseStorageAndNotifyClient`, try to use `storage.items.clear()`

```cs
private static void CloseStorageAndNotifyClient(InteractableStorage storage)
    {
        if (storage.isOpen)
        {
            if (storage.opener != null)
            {
                if (storage.opener.inventory.isStoring)
                {
                    storage.opener.inventory.closeStorageAndNotifyClient();
                }

                storage.opener = null;
            }

            storage.isOpen = !storage.isOpen;
        }
    }
```

**Comment by Fanya-dev** [comment]
The player at this point has his Inventory closed, because he must play the Point animation on the object

**Reply by sunnamed434** [reply]
So, just in case, what if someone tries to stay open the storage, for example when someone in vanish

**Reply by Fanya-dev** [reply]
No, I need the player to be able to use the vault after clear
